### PR TITLE
nagios: various improvements

### PR DESCRIPTION
**Pull request by symphorien**
* structured config for main config file allows to launch nagios in
debug mode without having to write the whole config file by hand
* build time syntax check
* all options have types (#76184) , one more example
* I find it misleading that the main nagios config file is linked in
/etc but that if you change the link in /etc/ and restart nagios, it
has no effect. Have nagios use /etc/nagios.cfg
* fix paths in example nagios config files, which allows to reuse it:
 ```
services.nagios.objectDefs =
   (map (x: "${pkgs.nagios}/etc/objects/${x}.cfg")
   [ "templates" "timeperiods" "commands" ]) ++ [ ./main.cfg ]
```
* for the above reason, add mailutils to default plugins

###### Things done

I tested by running nagios for a few days on 19.09.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @thoughtpolice @relrod

### CODE REVIEW

**Comment by aanderse** [comment]
@symphorien do you (or anyone) actually use `nagios` on NixOS?

**Comment by symphorien** [comment]
since a few days I have been setting one nagios instance on nixos, yes.

**Comment by aanderse** [comment]
Nice! I was evaluating it for a while but ended up going with `zabbix`.

The `nagios` module is pretty much abandoned at this point... would you be interested in becoming a maintainer for this module?

**Comment by symphorien** [comment]
I added myself as maintainer of the module.

File: nixos/modules/services/monitoring/nagios.nix
Comment (aanderse) [inline nixos/modules/services/monitoring/nagios.nix]:
```suggestion
        default = with pkgs; [ nagiosPluginsOfficial ssmtp mailutils ];
```

**Review by aanderse — CHANGES_REQUESTED** [review]
Thanks for listing yourself as a maintainer! I was recently thinking we should remove the module because it is so unmaintained!

This change looks good. Just one minor nitpick.

Do you have any motivation either in this PR or a future PR to write a NixOS test? This has always been a sticky point for me when making changes to the `nagios` module or packages in NixOS, and I always thought a NixOS module would be really cool  for `nagios` as it could demonstrate a fair bit.

Gosh now that I think about it... a `zabbix` test might be pretty cool too, right @mmahut :wink: (just razzing :smile:)

**Comment by symphorien** [comment]
A test that checks that alerts work might be easy to do, but not testing the web interface. I might try if I find the time and motivation, we'll see.

File: nixos/modules/services/monitoring/nagios.nix
Comment (dasJ) [inline nixos/modules/services/monitoring/nagios.nix]: Is it not possible to use `mkEnableOption` here?

**Comment by aanderse** [comment]
@symphorien this looks good. Can you possibly fixup the whitespace in the code generating `nagios-checked.cfg`? I'm interested in running this... then merging it. Do you have a minimal test example I could reproduce? Thanks!

File: nixos/modules/services/monitoring/nagios.nix
Comment (aanderse) [reply nixos/modules/services/monitoring/nagios.nix]: Yeah that is nice to do.

**Comment by symphorien** [comment]
@aanderse I fixed the spacing and add a simplistic nixos test.

**Review by aanderse — APPROVED** [review]
@symphorien you're a rock star! :rocket: Please squash into either 1 or 2 commits (if 2 then a single commit for the module and a single commit for the test, your call) and we'll merge this.

Thank you!

**Comment by symphorien** [comment]
done

**Comment by aanderse** [comment]
@GrahamcOfBorg test nagios

**Comment by aanderse** [comment]
@GrahamcOfBorg eval

**Pull request merged**
